public/chart.js

```diff
@@ -2,9 +2,29 @@
 var parseDate = d3.time.format("%Y-%m-%d").parse;
 
 function characterChart(svg, dataURL) {
+    var self = this;
+    var zoom = d3.behavior.zoom(); // Zooming behavior
+    this.fullYdomain = [];
+    this.fullXdomain = [];
+    this.prefix = ""; // CSV Path Prefix
+    this.resize = render; // Resizing behaviour
+    this.episodeData = null;
+    this.ready = 0;
+
+    // Get CSV prefix
+    (function () {
```

Comment:
It would be better coding style to declare this as a helper function somewhere and call it by name then.